Atheroprotective roles of smooth muscle cell phenotypic modulation and the TCF21 disease gene as revealed by single-cell analysis

Methods
1.Alignment: Fastq files from each experimental time point and mouse genotype were aligned to the reference genome individually using CellRanger Software (10x Genomics). Individual datasets were aggregated using the CellRanger aggr command without subsampling normalization. The aggregated dataset was then analyzed using the R package Seurat version 2.3.4 (refs. 34,35). 
2.Filtering: The dataset was trimmed of cells expressing fewer than 500 genes, and genes expressed in fewer than five cells. The number of genes, number of unique molecular identifiers and percentage of mitochondrial genes were examined to identify outliers. As an unusually high number of genes can result from a doublet event, in which two different cell types are captured together with the same barcoded bead, cells with >3,500 genes were discarded. Cells containing >7.5% mitochondrial genes were presumed to be of poor quality and were also discarded. 
3.Normalization: The gene expression values then underwent library-size normalization, in which raw gene counts from each cell were normalized relative to the total number of read counts present in that cell. The resulting expression values were then multiplied by 10,000 and log-transformed. 
4. HVG: Subsequent analyses were conducted using only the most highly variable genes in the dataset. Principal component analysis was used for dimensionality reduction, followed by clustering in principal component analysis space using a graph-based clustering approach. t-SNE was then used for two-dimensional visualization of the resulting clusters. 
5.Estimation of doublets: To estimate cell doublet rates, we used the baseline time point because the minimum time between tamoxifen gavage (tdT activation) and cell capture essentially excluded the possibility that transdifferentiation of SMCs to another cell type would affect the calculation. We determined the number of FACS-sorted tdT+ cells that had been assigned to cell clusters other than those known to express Myh11 at baseline (SMC1, SMC2, pericytes and a small number of phenotypically modulated SMCs). We then divided this number by the number of all tdT+ cells. Out of 3,707 tdT+ cells, 62 cells occurred in unexpected clusters, yielding a doublet rate of 62 out of 3,707 (1.7%). 
FASTQ files and matrices from scRNA-Seq data that support the findings of this study have been deposited in the Gene Expression Omnibus database with the primary accession code GSE131780.

		sample#
GSM3819834	ko_nonSMC_16wk_070218	18
GSM3819835	ko_nonSMC_16wk_070617	14
GSM3819836	ko_nonSMC_8wk_111417	10
GSM3819837	ko_SMC_16wk_070218	17
GSM3819838	ko_SMC_16wk_070617	13
GSM3819839	ko_SMC_16wk_121217	16
GSM3819840	ko_SMC_8wk_111417	9
GSM3819841	wt_nonSMC_16wk_061417	12
GSM3819842	wt_nonSMC_8wk_051717	6
GSM3819843	wt_nonSMC_8wk_112717	8
GSM3819844	wt_nonSMC_baseline_031617	2
GSM3819845	wt_nonSMC_baseline_120518	4
GSM3819846	wt_SMC_16wk_061417	11
GSM3819847	wt_SMC_16wk_121217	15
GSM3819848	wt_SMC_8wk_051717	5
GSM3819849	wt_SMC_8wk_112717	7
GSM3819850	wt_SMC_baseline_031617	1
GSM3819851	wt_SMC_baseline_12051	3

```{r}
library(Seurat)
library(ComplexHeatmap)
library(plyr)
```


```{r}
dge.heart.wirka <- read.table("~/storage/Others/Tcf21/Heart_Wirka/GSE131776_mouse_scRNAseq_wirka_et_al_GEO.txt.gz", header = TRUE, row.names = 1, sep = "\t")
saveRDS(dge.heart.wirka, file = "~/storage/Others/Tcf21/Heart_Wirka/dge.heart.wirka.rds")
#27086 cells
```

```{r}
dge.heart.wirka <- readRDS("~/storage/Others/Tcf21/Heart_Wirka/dge.heart.wirka.rds")
```

```{r}
samps <- unlist(lapply(colnames(dge.heart.wirka),function(x) unlist(strsplit(x,"[.]"))[2]))
```

```{r}
#18027 cells
dge.heart.wirka.wt <- dge.heart.wirka[,samps %in% c(12,6,8,2,4,11,15,5,7,1,3)]
```

```{r}
cells.genes <- apply(dge.heart.wirka.wt,2,function(x) sum(x>0))
genes.cells <- apply(dge.heart.wirka.wt,1,function(x) sum(x>0))
summary(cells.genes)
summary(genes.cells)
```

cells are already filtered.


```{r}
heart.wirka.wt <- CreateSeuratObject(raw.data = dge.heart.wirka.wt)
heart.wirka.wt <- NormalizeData(object = heart.wirka.wt, normalization.method = "LogNormalize", 
    scale.factor = 10000)
```

```{r}
heart.wirka.wt <- FindVariableGenes(object = heart.wirka.wt,mean.function = ExpMean, dispersion.function = LogVMR, x.low.cutoff = 0.0125, y.cutoff = 1, x.high.cutoff = 3)
heart.wirka.wt <- ScaleData(object = heart.wirka.wt)
heart.wirka.wt <- RunPCA(object = heart.wirka.wt, pc.genes = heart.wirka.wt@var.genes)
heart.wirka.wt <- RunTSNE(object = heart.wirka.wt, dims.use = 1:20)
```


```{r fig.width=6,fig.height=4.5}
PCAPlot(object = heart.wirka.wt, dim.1 = 1, dim.2 = 2)
PCAPlot(object = heart.wirka.wt, dim.1 = 1, dim.2 = 3)
PCElbowPlot(object = heart.wirka.wt)
```

```{r fig.width=6,fig.height=4.5}
TSNEPlot(object = heart.wirka.wt)
```


```{r}
heart.wirka.wt <- FindClusters(object = heart.wirka.wt, reduction.type = "pca", dims.use = 1:20, save.SNN = TRUE, resolution = 0.4)
```


```{r fig.width=6,fig.height=4.5}
DimPlot(object = heart.wirka.wt, reduction = "tsne", group.by = "res.0.4",do.label = TRUE)
```
```{r}
heart.wirka.wt <- SetIdent(heart.wirka.wt, ident.use = heart.wirka.wt@meta.data$res.0.4)
```

```{r}
saveRDS(heart.wirka.wt, "~/storage/Others/Tcf21/Heart_Wirka/heart.wirka.wt.rds")
heart.wirka.wt <- readRDS("~/storage/Others/Tcf21/Heart_Wirka/heart.wirka.wt.rds")
```


```{r}
heart.wirka.markers <- list()
for(i in c(0:14)){
  print(i)
  tmp <- FindMarkers(heart.wirka.wt, ident.1 = i, min.pct = 0.2, logfc.threshold = 1,only.pos = TRUE)
  tmp <- tmp[tmp$p_val_adj < 0.01, ]
  print(head(tmp))
  heart.wirka.markers[[i+1]] <- tmp
}
```

By comparing with the markers in the supplmental table, we can get the following correspondence:
0: SMC1
1: SMC2
2: Fibroblast
3: Fibroblast 2
4: Modulated SMC
5: Fibroblast
6: Macrophage
7: Endothelial 1
8: Endothelial 1
9: Pericyte
10: Endothelial 2
11: Tcell
12: Epithelial-like
13: Neuron
14: None

```{r}
heart.wirka.wt@meta.data$celltypes <- mapvalues(heart.wirka.wt@meta.data$res.0.4, from = c(0:14), to = c("SMC","SMC","Fibroblast","Fibroblast_2","Modulated_SMC","Fibroblast","Macrophage","Endothelial_1","Endothelial_1","pericyte","Endothelial_2","Tcell","Epithelial","Neuron","None"))
```

```{r}
cell.types <- unique(heart.wirka.wt@meta.data$celltypes)
cell.types <- cell.types[cell.types!="None"]
```

```{r}
dge.heart.wirka.wt.norm <- apply(heart.wirka.wt@raw.data,2,function(x) x/sum(x)*1e4)

heart.wirka.cen <- c()
for(i in cell.types){
  cells <- heart.wirka.wt@cell.names[heart.wirka.wt@meta.data$celltypes==i]
  cen <- apply(dge.heart.wirka.wt.norm[,cells],1,mean)
  heart.wirka.cen <- cbind(heart.wirka.cen, cen)
}
colnames(heart.wirka.cen) <- cell.types
```

```{r}
mou.cen <- read.table("~/storage/HumanMacaqueMouseCompAnaysis/ligandReceptor/MouseAdultST25_7SomaticCellTypes_Centroids_Qianyi.txt",header = TRUE, row.names = 1)
colnames(mou.cen) <- c("InnateLymphoid","Macrophage","Endothelial", "Myoid", "Leydig","Sertoli","Tcf21+")
mou.markers <- read.table("~/storage/Others/Tcf21/MouseSomaMarkers.txt", header = TRUE)
mou.soma.markers <- intersect(unique(mou.markers$gene), rownames(mou.cen))
```

```{r}
genes.int <- intersect(rownames(dge.heart.wirka.wt.norm), rownames(mou.cen))
markers.int <- intersect(rownames(dge.heart.wirka.wt.norm), mou.soma.markers)
```

```{r}
mou.germ.heart.wirka.cor <- cor(mou.cen[genes.int,], heart.wirka.cen[genes.int,], method = "spearman")
mou.germ.heart.wirka.marker.cor <- cor(mou.cen[markers.int,], heart.wirka.cen[markers.int,], method = "spearman")
#saveRDS(mou.germ.heart.wirka.cor, "~/storage/Others/Tcf21/mou.germ.heart.wirka.cor.rds")
```

```{r fig.width=5, fig.height=4.5}
pdf("~/storage/Others/Tcf21/Cardio_wirka.pdf", width = 5, height = 4.5)
#heart.clusters <- c("Fibroblast","Fibroblast_2","Modulated_SMC", "SMC","pericyte","Endothelial_1","Endothelial_2","Epithelial","Neuron", "Macrophage","Tcell")
Heatmap(mou.germ.heart.wirka.cor, cluster_rows = TRUE, cluster_columns = TRUE, show_row_dend = FALSE, show_column_dend = FALSE, column_title = "mouse tesits somatic vs. Cardio_wirka")
dev.off()
```

```{r fig.width=5, fig.height=4.5}
pdf("~/storage/Others/Tcf21/Cardio_wirka_marker.pdf", width = 5, height = 4.5)
#heart.clusters <- c("Fibroblast","Fibroblast_2","Modulated_SMC", "SMC","pericyte","Endothelial_1","Endothelial_2","Epithelial","Neuron", "Macrophage","Tcell")
Heatmap(mou.germ.heart.wirka.marker.cor, cluster_rows = TRUE, cluster_columns = TRUE, show_row_dend = FALSE, show_column_dend = FALSE, column_title = "mouse tesits somatic vs. Cardio_wirka")
dev.off()
```

```{r}
save.image(file = "~/storage/Others/Tcf21/Heart_Wirka/bak20200405.RData")
```

```{r}
load(file = "~/storage/Others/Tcf21/Heart_Wirka/bak20200405.RData")
```

```{r}
saveRDS(heart.wirka.cen, file = "~/storage/Others/Tcf21/heart.wirka.cen.rds")
```


Separate by healthy and injured tissue
```{r}
samps <- unlist(lapply(colnames(heart.wirka.wt@data),function(x) unlist(strsplit(x,"[.]"))[2]))
heart.wirka.wt@meta.data$baseline = samps %in% c(1:4)
```

```{r}
heart.wirka.healthy.cen <- c()
for(i in cell.types){
  cells <- heart.wirka.wt@cell.names[heart.wirka.wt@meta.data$celltypes==i & heart.wirka.wt@meta.data$baseline==TRUE]
  cen <- apply(dge.heart.wirka.wt.norm[,cells],1,mean)
  heart.wirka.healthy.cen <- cbind(heart.wirka.healthy.cen, cen)
}
colnames(heart.wirka.healthy.cen) <- cell.types

heart.wirka.injured.cen <- c()
for(i in cell.types){
  cells <- heart.wirka.wt@cell.names[heart.wirka.wt@meta.data$celltypes==i & heart.wirka.wt@meta.data$baseline==FALSE]
  cen <- apply(dge.heart.wirka.wt.norm[,cells],1,mean)
  heart.wirka.injured.cen <- cbind(heart.wirka.injured.cen, cen)
}
colnames(heart.wirka.injured.cen) <- cell.types
```

```{r}
heart.wirka.healthy.cen[,"Modulated_SMC"] <- NA
saveRDS(heart.wirka.healthy.cen, file = "~/storage/Others/Tcf21/heart.wirka.healthy.cen.rds")
saveRDS(heart.wirka.injured.cen, file = "~/storage/Others/Tcf21/heart.wirka.injured.cen.rds")
```